Add CNPG cluster chart as vshnpostgresql

mikeshootzz opened the pull request:
#### What this PR does / why we need it:

- Add Cloud Native PG "cluster" helmchart from https://github.com/cloudnative-pg/charts
- Extend helmchart with backups using the barman cloud plugin

#### Checklist

- [x] Chart Version bumped
- [x] I have run `make docs`
- [x] Variables are documented in the values.yaml using the format required by [Helm-Docs](https://github.com/norwoodj/helm-docs#valuesyaml-metadata).
- [x] PR contains the label that identifies the chart, e.g. `chart/<chart-name>`
- [x] PR contains the label that identifies the type of change, which is one of
      [ `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency` ]

mikeshootzz committed:
Add files from upstream chart

This copies the upstream helmchart for provisioning a cnpg cluster as-is
without any modifications.

mikeshootzz committed:
Add Apache-2.0 license for vshnpostgresql chart

Since the upstream helmcharts for cloud native pg are licensed under the
Apache-2.0 license, we have to give special notice for this chart and
use the same license accordingly.

mikeshootzz committed:
Add Barman Cloud Plugin configuration

This adds the barman cloud plugin as a backup option for cnpg.

mikeshootzz committed:
Docs: Align documentation with vshnmariadb

This aligns the documentation of vshnpostgresql with vshnmariadb

zugao commented on charts/vshnpostgresql/templates/barman-cloud-objectstore.yaml:
```diff
@@ -0,0 +1,55 @@
+apiVersion: barmancloud.cnpg.io/v1
+kind: ObjectStore
+metadata:
+  name: {{ include "cluster.backups.objectStoreName" . }}
```
let's craft the name based on the release name

zugao commented on charts/vshnpostgresql/templates/barman-plugin-s3-creds.yaml:
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ default (printf "%s-barman-plugin-s3-creds" (include "cluster.fullname" .)) .Values.backups.secret.name }}
```
here as well

zugao commented on charts/vshnpostgresql/templates/ca-bundle.yaml:
```diff
@@ -1,10 +1,14 @@
 kind: Secret
 metadata:
+  {{- if eq .Values.backups.method "plugin" }}
+  name: {{ .Values.backups.endpointCA.name | default (printf "%s-barman-plugin-ca-bundle" (include "cluster.fullname" .)) | quote }}
```
here as well

zugao reviewed (changes requested):
I think the name should be changed

mikeshootzz committed:
Base names on relesae name instead of cluster fullname

This uses the release name instead of the cluster fullname as a prefix
for the barman object store related resources.

Kidswiss commented on charts/vshnpostgresql/templates/barman-cloud-objectstore.yaml:
```diff
@@ -0,0 +1,55 @@
+  name: {{ include "cluster.backups.objectStoreName" . }}
+  namespace: {{ include "cluster.namespace" . }}
+  labels: {{ include "cluster.labels" . | nindent 4 }}
+spec:
```
Probably doesn't make sense to export everything that the CRD has via the chart. Let's just implement what we need for the time being.

mikeshootzz committed:
Simplify object store configuration

This simplifies the object store and its related resources to only
expose what we actually need

mikeshootzz commented:
We'll wait for the upstream implementation to be finished: https://github.com/cloudnative-pg/charts/pull/680 and publish the upstream PR's chart to our GitHub registry.
